homeassistant/components/matter/adapter.py

```diff
@@ -142,10 +138,18 @@ def node_removed_callback(event: EventType, node_id: int) -> None:
     def _setup_node(self, node: MatterNode) -> None:
         """Set up an node."""
         LOGGER.debug("Setting up entities for node %s", node.node_id)
-
-        for endpoint in node.endpoints.values():
-            # Node endpoints are translated into HA devices
-            self._setup_endpoint(endpoint)
+        try:
+            for endpoint in node.endpoints.values():
+                # Node endpoints are translated into HA devices
+                self._setup_endpoint(endpoint)
+        except Exception as err:  # noqa: BLE001
```

Comment:
We generally only allow catching bare exceptions in the config flow, can we be more specific?

Reply:
I think it's ok in this case since we don't know what exception it would be and we'll log it at exception level.

 It's a guard to not crash the whole integration if one device brings some unexpected endpoint information.

It would be good to add a test though.

Reply:
Yeah, exactly my thoughts. Lets at least try to setup the rest of the integration.
Indeed adding a test would be nice.